sw/device/mask_rom/boot.md

```diff
@@ -0,0 +1,110 @@
+# Pseudo-code for Secure Boot Process.
+
+This file should be read in conjunction with secure boot specification. Some
+references to that document are included.
+
+
+1. Power on (entirely in hardware) Open Q: Whether SW has to configure PMP
+  initial region.
+
+2. Execution Begins
+  - CRT Startup Code (ASM)
+    - Clean Slate (SRAM, Registers)
+    - Well-defined interrupt/exception handler
+        Open Q: Do we want to enable interrupts in mask rom?
+    - Jump into C code (`boot`):
+
+```c
+void boot(void) {
+  // - Clean all other hardware state (using DIF Reset functions)
+  clean_device_state();
+
+  // - Chip-specific startup functionality
+  //   - Clocks
+  //   - AST
+  //   - Flash
+  //   - Entropy
+  //   Open Q: Proprietary Software Strategy.
+  chip_specific_startup();
```

Comment:
Is this chip specific start up where we might do things like ROM boot-strap if it is required?
There's been some recent discussions (not widely circulated yet) on how ROM might be able to do this in order to conserve overall pins.  This could be something we want to discuss more.

Reply:
I discussed this with @moidx yesterday, and the answer we reached is that i see `chip_specific_startup` as dealing with peripheral initialization which is chip-specific. In the update PR, I've added a bootstrap phase that comes immediately after `chip_specific_startup` and before `read_boot_policy`.